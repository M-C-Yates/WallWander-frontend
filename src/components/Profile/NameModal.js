import React from 'react';

export const NameModal = (props) => {
	const { hide, setName, visible } = props;

	// visible ? show() : hide();
	if (visible) {
		return (
			<div className="modal">
				<div className="modal-overlay" />

				<form className="form modal-form" onSubmit={props.submit} method="post">
					<button className="modal-close" onClick={() => hide()}>
						X
					</button>
					<div className="modal-form-box">
						<label className="form-label" htmlFor="name">
							Username
						</label>
						<input
							type="text"
							placeholder="min 6 characters"
							className="form-input"
							onChange={(e) => setName(e.target.value)}
							minLength="4"
						/>
					</div>
					<button className="btn btn-form">Submit</button>
				</form>
			</div>
		);
	} else {
		return <React.Fragment />;
	}
};

export default NameModal;
