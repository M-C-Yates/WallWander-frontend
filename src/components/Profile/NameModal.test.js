import React from 'react';
import { render, cleanup } from '@testing-library/react';

import NameModal from './NameModal';

afterEach(cleanup);

describe('NameModal', () => {
	it('should match snapshot', () => {
		const { asFragment } = render(<NameModal />);
		expect(asFragment).toMatchSnapshot();
	});
});
