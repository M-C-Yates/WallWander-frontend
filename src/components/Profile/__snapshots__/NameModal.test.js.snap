// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameModal should match snapshot 1`] = `[Function]`;
